netutils/ftpc/ftpc_transfer.c

```diff
@@ -247,7 +247,10 @@ static FAR char *ftpc_abspath(FAR struct ftpc_session_s *session,
 
       else if (relpath[1] == '/')
         {
-          asprintf(&ptr, "%s%s", homedir, &relpath[1]);
+          if (asprintf(&ptr, "%s%s", homedir, &relpath[1]) < 0)
+            {
+              ptr = NULL;
```

Comment:
why this is needed?

Reply:
It is already set to NULL at line 224

Reply:
As said, the contents of the pointer are undefined if `asprintf` fails.

So `asprintf` can set a value to this pointer, and *then* fail. In this case it is not required for it to restore it to NULL. Thus the pointer now contains garbage.

If you need to be sure that the pointer is `NULL` after an `asprintf` failure, then it must be explicitly set so, *after* the `asprintf` call.

Reply:
Ok. Since this is the application code and ptr is used later, let's keep applications compatible with other systems.